src/content/blog/astro-4140.mdx

```diff
@@ -0,0 +1,265 @@
+
+```ts
+// src/content/config.ts
+import { defineCollection } from "astro:content";
+import { feedLoader } from "@ascorbic/feed-loader";
+
+const podcasts = defineCollection({
+  loader: feedLoader({
+    url: "https://feeds.99percentinvisible.org/99percentinvisible",
+  }),
+});
+
+export const collections = { podcasts };
+```
+
+### Learn more
+
+To find out more about using the Content Layer API, check out [the Content Layer RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
+
+## Experimental: Intellisense inside content
+
+Astro 4.14 introduces experimental support for intellisense inside the frontmatter of your content. This feature helps you write content files more efficiently by providing completions, validation, hover information and more for frontmatter keys and values, based on your content schemas, directly in your editor.
+
+This feature is available in VS Code and other editors that support the Language Server Protocol for Markdown, MDX, and Markdoc files.
+
+<BlogContentImage src={frontmatterIntellisense} />
+
+To enable content file intellisense, add the following to your Astro config:
+
+```js
+import { defineConfig } from 'astro';
+
+export default defineConfig({
+	experimental: {
+		contentIntellisense: true
+	}
+})
+```
+
+and in your editor, enable the `astro.content-intellisense` setting, or for other editors, pass `contentIntellisense: true` to the initialization parameters of the Astro language server.
+
+```json
+{
+	"astro.content-intellisense": true
+}
+```
+
+This feature currently only works in the editor, and will be enabled in `astro check` in a future release.
+
+## Deprecate support for dynamic `prerender` values
+
+In order to improve our bundling and tree-shaking capabilities, Astro 4.14 deprecates support for dynamic `prerender` values. This means that you can no longer use dynamic values for the `prerender` option in your Astro pages.
+
+This feature can now be replaced with the new `astro:route:setup` hook in Astro integrations, which allows you to dynamically control the prerendering of specific pages.
+
+```diff
+---
+- export const prerender = import.meta.env.PROD;
+---
+```
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  integrations: [setPrerender()],
+});
+
+function setPrerender() {
+  return {
+    name: 'set-prerender',
+    hooks: {
+      'astro:route:setup': ({ route }) => {
+        if (route.component.endsWith('/blog/[slug].astro')) {
+          route.prerender = true;
+        }
+      },
+    },
+  };
+}
+```
+
+## New `injectTypes` integration API
+
+Type generation has become all the rage in recent years, and Astro is no exception as we use type generation pretty extensively for features like Content Collections (or, now, Content Layer), or Astro DB.
```

Comment:
```suggestion
Type generation has become all the rage in recent years, and Astro is no exception! We use type generation pretty extensively for features like Astro DB and Content Collections (with the new Content Layer API that you're definitely going to enable as soon as you're done reading this post).
```
Not sure we've decided to imply that "content collections are no longer a thing" but here's another way if you wanted to inject some personality into this line!